File: scripts/count_md_files.py
#!/usr/bin/env python3
"""Count markdown (`*.md`) files under a directory.

Python port of `scripts/count_md_files.ts`, kept in sync byte-for-byte
semantics: recursive walk, count files whose name ends in `.md`, print the
integer count on stdout. Used by the Python CI workflow's
"Docs scan baseline" step.
"""
from __future__ import annotations

import os
import sys
from pathlib import Path


def count_markdown_files(root: Path) -> int:
    count = 0
    for _dirpath, _dirnames, filenames in os.walk(root):
        for name in filenames:
            if name.endswith(".md"):
                count += 1
    return count


def main(argv: list[str]) -> int:
    if len(argv) != 1:
        print(
            "Usage: python3 scripts/count_md_files.py <directory>",
            file=sys.stderr,
        )
        return 1
    target = Path(argv[0]).resolve()
    if not target.exists():
        print(f"error: directory does not exist: {target}", file=sys.stderr)
        return 1
    if not target.is_dir():
        print(f"error: path is not a directory: {target}", file=sys.stderr)
        return 1
    print(count_markdown_files(target))
    return 0


if __name__ == "__main__":
    raise SystemExit(main(sys.argv[1:]))

File: scripts/count_md_files.ts
import * as fs from 'node:fs';
import * as path from 'node:path';
import * as process from 'node:process';

function countMarkdownFiles(dir: string): number {
  let count = 0;
  function walk(d: string) {
    for (const f of fs.readdirSync(d)) {
      const full = path.join(d, f);
      const stat = fs.statSync(full);
      if (stat.isDirectory()) {
        walk(full);
      } else if (f.endsWith('.md')) {
        count++;
      }
    }
  }
  walk(dir);
  return count;
}

function main() {
  const args = process.argv.slice(2);
  if (args.length !== 1) {
    console.error('Usage: node --experimental-strip-types scripts/count_md_files.ts <directory>');
    process.exit(1);
  }
  const target = path.resolve(args[0]);
  if (!fs.existsSync(target)) {
    console.error(`error: directory does not exist: ${target}`);
    process.exit(1);
  }
  if (!fs.statSync(target).isDirectory()) {
    console.error(`error: path is not a directory: ${target}`);
    process.exit(1);
  }
  console.log(countMarkdownFiles(target));
  process.exit(0);
}

main();
